### DISCUSSION TITLE
how to define project owned typescript declaration

### DISCUSSION THREAD

**Discussion by honwhy** [wxt-dev/wxt#403] [Q&A]
### Describe the bug

failed to define project owned typescript declaration

### To Reproduce

in content script, import vue sfc file, and in sfc file import 'turndown'

[turndown](https://github.com/mixmark-io/turndown)
try to declare module, and add to `<rootDir>/tsconfig.json`
```
export {}

declare module 'turndown' {
  export class TurndownService {}
}

```
in `<rootDir>/tsconfig.json`,
```
{
  "extends": "./.wxt/tsconfig.json"
+ "include": ["./global.d.ts"]
}
```
but results in more typescript errors
### Screenshots

![image](https://github.com/wxt-dev/wxt/assets/2212273/66b4189b-209e-422e-8399-c83573cb809b)

**Comment by aklinker1** [comment]
TLDR: Remove `export {}` from your declaration file and remove `"include"` from your `tsconfig.json`, then everything will work.

---

When your TSConfig files extends another and you set the `include` or `exclude` options, the extended `include` and `exclude` options are overridden, not merged.

So in your case, you're telling typescript to only include your global file in the project, and none of your source code. Look at `.wxt/tsconfig.json` for what `include` defaults to for your project.

As for the `export {}`, this makes a declaration file a module, not a declaration file, and that means you have to import it before you get any of the declarations you defined in it.

In general, you don't add `export {}` when declaring modules.

**Comment by nnx0r** [comment]
Move to discussion, I'd say. Someone else may have same/similar question in the future.

**Comment by PatrykKuniczak** [comment]
Maybe this can be helpful if anybody'll have similar problem in the future:
https://github.com/wxt-dev/wxt/pull/1981
